src/main/java/uk/gov/hmcts/reform/unspec/service/BusinessProcessService.java

```diff
@@ -0,0 +1,48 @@
+package uk.gov.hmcts.reform.unspec.service;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.unspec.callback.CaseEvent;
+import uk.gov.hmcts.reform.unspec.model.BusinessProcess;
+import uk.gov.hmcts.reform.unspec.stateflow.model.State;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import static uk.gov.hmcts.reform.unspec.model.BusinessProcessStatus.FINISHED;
+import static uk.gov.hmcts.reform.unspec.model.BusinessProcessStatus.READY;
+
+@Service
+@RequiredArgsConstructor
+public class BusinessProcessService {
+
+    private static final String ERROR_MESSAGE = "Business Process Error";
+
+    private final ObjectMapper mapper;
+
+    public List<String> updateBusinessProcess(Map<String, Object> data, CaseEvent caseEvent) {
+        return updateBusinessProcess(data, caseEvent, null);
+    }
+
+    public List<String> updateBusinessProcess(Map<String, Object> data, CaseEvent caseEvent, State stateFlowState) {
+        BusinessProcess businessProcess = mapper.convertValue(data.get("businessProcess"), BusinessProcess.class);
+        if (hasNoOngoingBusinessProcess(businessProcess)) {
+            data.put("businessProcess", BusinessProcess.builder().camundaEvent(caseEvent.name()).status(READY).build());
+            Optional.ofNullable(stateFlowState)
+                .map(State::getName)
+                .ifPresent(stateName -> data.put("stateFlowState", stateName));
```

Comment:
Maybe just a suggestion to call this `Flow State`.

Reply:
This is actually quite important. The flow state should never be stored in the case data but always computed on the fly. This is to make it easier once we will be live, to change the state flow engine rules without have to also patch the data.

Reply:
removed as discussed